### ISSUE TITLE
SearchField Component doesn't trigger onSearch callback function on clearing the search field

### ISSUE THREAD

**Issue by nagrawal3** [cerner/terra-core#541]
# Issue Description

SearchFiled triggers the onSearch callback when searchText is either greater than or equal to the minimumSearchTextLength and doesn't trigger when user clears the searchText filed.

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [X] Bug
- [ ] Other

## Expected Behavior

onSearch callback function should be called when either the length of searchText is greater than or equal to the minimumSearchTextLength, or when searchText is cleared.

## Current Behavior

onSearch callback function is not being triggered when user clear the searchText field

## Environment

* Component Version: 1.0.0
* Browser Name and Version: Safari 10.1.1
* Operating System and version (desktop or mobile): Mac OS X El Capitan 10.11.6

**Comment by nagrawal3** [comment]
one possible solution: this line should check if the searchText.length is either equal to 0 or >= minimumSearchTextLength

**Comment by dkschoonover** [comment]
It might make sense to expose a separate callback - as a consumer I wouldn't expect onSearch to be called without search text.  How would people feel about exposing an onSearchClear callback that is called anytime search text goes below the minimum search length?

I can contribute this change if people think that would be okay.

**Comment by tbiethman** [comment]
@nagrawal3 Are you using this for some sort of filter? Would it make sense for you to set the minimumSearchTextLength prop on the SearchField to zero? That would ensure you get every callbacks for every change to the search text.

**Comment by nagrawal3** [comment]
@tbiethman we are using it for the filter. We can use minimumSearchTextLength as 0 and then add filter accordingly to the searchText return by the Search Field callback.

**Comment by emilyrohrbough** [comment]
# Tech Design For `onInvalidSearch()`
## Summary:
Adding an `onInvalidSearch()` callback function would provide users with the option to opted-in for callback for invalid search entry. Invalid search entry would include any `searchText` less than the indicated `minimumSearchTextLength`. 

Note: the default `minimumSearchTextLength` is 2, so providing an `onInvalidSearch()` would also act as a clear handler. 

## Add React Prop:
| Prop     | Type | Default | Description  |
|----------|------|---------|--------------|
| onInvalidSearch | func | undefined    | A callback to indicate an invalid search. Sends parameter {String} searchText. |

## Example:
```
import React from 'react';
import SearchField from '../../lib/SearchField';

class CallbackSearchField extends React.Component {

  constructor(props) {
    super(props);
    this.state = { searchText: '', message: '', };
    this.handleSearch = this.handleSearch.bind(this);
    this.handleInvalidSearch = this.handleInvalidSearch.bind(this);
  }

  handleSearch(searchText) {
    this.setState({ searchText, message: 'Search Text: ' });
  }

  handleInvalidSearch(searchText) {
    this.setState({ searchText, message: 'INVALID Search Text: ' });
  }

  render() {
    return (
      <div>
        <h3> Minimum Search Length is 5 </h3>
        <SearchField id="searchfield" onSearch={this.handleSearch} onInvalidSearch={this.handleInvalidSearch} minimumSearchTextLength={5} />
        <div id="search-callback-text">
          {this.state.message}{this.state.searchText}
        </div>
      </div>
    );
  }

}

export default CallbackSearchField;
```

**Comment by dkasper-was-taken** [comment]
+1

**Comment by mjhenkes** [comment]
+1

**Comment by Matt-Butler** [comment]
+1

**Comment by bjankord** [comment]
+1

**Comment by bjankord** [comment]
This has been released in terra-search-input@1.2.0
